confgenerator/logging_processors.go

```diff
@@ -39,77 +44,182 @@ func init() {
 	LoggingProcessorTypes.RegisterType(func() Component { return &LoggingProcessorParseJson{} })
 }
 
-// A LoggingProcessorParseRegex applies a regex to the specified field, storing the named capture groups as keys in the log record.
-type LoggingProcessorParseRegex struct {
-	ConfigComponent             `yaml:",inline"`
-	LoggingProcessorParseShared `yaml:",inline"`
+// A LoggingProcessorParseRegexSimple applies a regex to the specified field, storing the named capture groups as keys in the log record.
+// This was maintained in addition to the parse_regex_complex to ensure backward compatibility with any existing configurations
+type LoggingProcessorParseRegexSimple struct {
```

Comment:
Can you please keep type name as `LoggingProcessorParseRegex` since the receiver type is still `parse_regex` (and we can't change that)?